lib/finch.ex

```diff
@@ -83,21 +83,34 @@ defmodule Finch do
     for any unspecified URLs. See "Pool Configuration Options" below for details on the possible
     map values. Default value is `%{default: [size: #{@default_pool_size}, count: #{@default_pool_count}]}`.
 
+    * `:request_transformer` - A callback function to modify Requests. The callback is useful for
+    injecting distributed tracing headers. The value should be set to a function that accepts and
+    returns a `%Finch.Request{}`. Attention should be paid to performance as this function gets
+    called during every request.
+
```

Comment:
Not sure whether this is simpler or more clear, but just a suggested re-wording:
```suggestion
    * `:request_transformer` - A callback that can be used to modify Requests as they're being
    made (for example, to inject distributed tracing headers. The arity-1 function will receive the `t:Finch.Request.t/0` and must also return the transformed `t:Finch.Request.t/0`. Note that this
    function will be called synchronously during every request, so care should be taken to ensure
    that it does not introduce unnecessary latency.
```